core/templates/domain/topic/topic-object.model.ts

```diff
@@ -584,40 +584,47 @@ export class Topic {
   // Reassigns all values within this topic to match the existing
   // topic. This is performed as a deep copy such that none of the
   // internal, bindable objects are changed within this topic.
-  copyFromTopic(otherTopic: Topic): void {
-    this._id = otherTopic.getId();
-    this.setName(otherTopic.getName());
-    this.setAbbreviatedName(otherTopic.getAbbreviatedName());
-    this.setUrlFragment(otherTopic.getUrlFragment());
-    this.setThumbnailFilename(otherTopic.getThumbnailFilename());
-    this.setThumbnailBgColor(otherTopic.getThumbnailBgColor());
-    this.setDescription(otherTopic.getDescription());
-    this.setLanguageCode(otherTopic.getLanguageCode());
-    this.setPracticeTabIsDisplayed(otherTopic.getPracticeTabIsDisplayed());
-    this.setMetaTagContent(otherTopic.getMetaTagContent());
-    this.setPageTitleFragmentForWeb(otherTopic.getPageTitleFragmentForWeb());
-    this.setSkillSummariesForDiagnosticTest(
-      otherTopic.getSkillSummariesForDiagnosticTest());
-    this._version = otherTopic.getVersion();
-    this._nextSubtopicId = otherTopic.getNextSubtopicId();
-    this.clearAdditionalStoryReferences();
-    this.clearCanonicalStoryReferences();
-    this.clearUncategorizedSkills();
-    this.clearSubtopics();
-
-    this._canonicalStoryReferences = otherTopic.getCanonicalStoryReferences();
-    this._additionalStoryReferences =
+  copyFromTopic(otherTopic: Topic): Topic {
```

Comment:
This is a good refactor, but if we're going with this route then it would be better to frame this as createCopyOfTopic() instead. You don't need to pass any arguments because the topic that's being copied would be the current object (i.e. `this`). The method would then return a new topic that's a copy of the current topic. The docstring should also be updated.

Reply:
Done